components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -553,61 +560,133 @@ private HashMap getAPIPermissionArray(String permissionJsonString) throws ParseE
+                workflow.setTransitionState(status);
+
+                workflow.setWorkflowReference(originalAPI.getId());
+                workflow.setExternalWorkflowReference(UUID.randomUUID().toString());
+                workflow.setCreatedTime(LocalDateTime.now());
+                workflow.setWorkflowType(WorkflowConstants.WF_TYPE_AM_API_STATE);             
+                workflow.setInvoker(getUsername());
+                
+                //setting attributes for internal use. These are set to use from outside the executor's method
+                //these will be saved in the AM_WORKFLOW table so these can be retrieved later
+                workflow.setAttribute(WorkflowConstants.ATTRIBUTE_API_CUR_STATE, currentState.getState());
+                workflow.setAttribute(WorkflowConstants.ATTRIBUTE_API_TARGET_STATE, status);
+                workflow.setAttribute(WorkflowConstants.ATTRIBUTE_API_LC_INVOKER, getUsername());
+                workflow.setAttribute(WorkflowConstants.ATTRIBUTE_API_LAST_UPTIME,
+                        originalAPI.getLastUpdatedTime().toString());
+                
+                workflowResponse = executor.execute(workflow);             
+                workflow.setStatus(workflowResponse.getWorkflowStatus());
+
+                addWorkflowEntries(workflow);
+
+                if (WorkflowStatus.APPROVED == workflowResponse.getWorkflowStatus()) {
+                    completeWorkflow(executor, workflow);
+                } else {
+                    getApiDAO().updateAPIWorkflowStatus(api.getId(), APILCWorkflowStatus.PENDING);
+                }
+            } else if (api != null && APILCWorkflowStatus.PENDING.toString().equals(api.getWorkflowStatus())) {
+                String message = "Pending state transition for api :" + api.getName();
+                log.error(message);     
+                throw new APIManagementException(message, ExceptionCodes.WORKFLOW_PENDING);
+            } else {
+                throw new APIMgtResourceNotFoundException("Requested API " + apiId + " Not Available");
+            }
+        } catch (APIMgtDAOException e) {
+            String errorMsg = "Couldn't change the status of api ID " + apiId;
+            log.error(errorMsg, e);
+            throw new APIManagementException(errorMsg, e, ExceptionCodes.APIMGT_DAO_EXCEPTION);
+        } catch (LifecycleException e) {
+            String errorMsg = "Couldn't change the status of api ID " + apiId;
+            log.error(errorMsg, e);
+            throw new APIManagementException(errorMsg, e, ExceptionCodes.APIMGT_LIFECYCLE_EXCEPTION);
+        }
+        return workflowResponse;
+    }
+    
+    private void updateAPIStatusForWorkflowComplete(String apiId, String status, String updatedBy, LocalDateTime time)
+            throws APIManagementException {
+        boolean requireReSubscriptions = false;
+        boolean deprecateOlderVersion = false;
+        try {
```

Comment:
The default value is false. Hence we don't need to initialize this to false.  Shall we change this?